scripts/rollup-plugin-dependency-condition.js

```diff
@@ -0,0 +1,17 @@
+import { createRequire } from "module";
+
+const require = createRequire(import.meta.url);
+
+export default function (value) {
+  return {
+    name: "dependency-condition",
+    resolveId(specifier, referrer) {
+      const re = /^(.*?-BABEL_8_BREAKING)-(?:true|false)$/;
+      const match = specifier.match(re);
+      if (!match) return null;
+      return require.resolve(`${match[1]}-${!!value}`, {
+        paths: [referrer],
+      });
+    },
+  };
+}
```

Comment:
How about
```js
export default function (value) {
  const re = value
    ? /^(.*?-BABEL_8_BREAKING)-false$/
    : /^(.*?-BABEL_8_BREAKING)-true$/;
  return {
    name: "dependency-condition",
    resolveId(source, importer, options) {
      const match = source.match(re);
      if (!match) return null;
      return this.resolve(`${match[1]}-${!!value}`, importer, options);
    },
  };
}
```
?
So that we can pass through the resolve options, which may have been enhanced by rollup, and the plugin should also work with other plugins that depend on `resolveId`, since otherwise they would be short-circuited.